cfme/tests/cloud/test_providers.py

```diff
@@ -484,6 +485,42 @@ def test_select_key_pair_none_while_provisioning(appliance, request, has_no_clou
     assert view.form.properties.guest_keypair.read() == '<None>'
 
 
+@pytest.mark.tier(3)
+@pytest.mark.provider([AzureProvider], override=True)
+def test_password_requirements(appliance, request, has_no_cloud_providers, setup_provider):
+    """
+        Requirement: Have an Azure provider
+        1. Compute -> Cloud -> Instances
+        2. Click on Provision Instances in Toolbar
+        3. Select template.
+        4. Go to Customisation, fill password that doesn't match the criteria:
+            * must be 12-72 characters
+            * have 3 of the following - one lowercase character, one uppercase character,
+              one number and one special character
+        5. Error message should be displayed.
+    """
+    view = navigate_to(appliance.collections.cloud_instances, 'Provision')
+    view.image_table[0].click()
+    view.form.continue_button.click()
+    message = (
+        "'Customize/Password' must be correctly formatted. The password must be 12-72 characters, "
+        "and have 3 of the following - one lowercase character, one uppercase character, "
+        "one number and one special character.")
+
+    view.form.customize.fill({
+        "admin_username": "some_value",
+    })
+
+    for pw in ("abcdefghijkl_",
+               "ABCDEFGHIJKL_",
+               "ABCDEFGHIJKLa",
+               "abcdefgh_1A"):
+        view.form.customize.fill({"root_password": pw})
+        view.form.submit_button.click()
+        wait_for(lambda: message in (m.text for m in view.flash.messages),
```

Comment:
Question: why we need to use wait_for here, as we can use soft_assert(view.flash.assert_messeges(message)) here to check all options?

Reply:
We need to wait for the messages to appear.

Reply:
@jaryn Are you intending to make the above changes to the test? Thoughts? Marking WIPTEST until I hear back :)

Reply:
I will change the code to use `m.read()` instead of accessing `m.text` if you think it is better. Although the result is equivalent, so I don't really see a reason.

`m.dismiss()` is not needed. The messages are dismissed upon IIRC `submit_button.click()` method call, but it feels better to dismiss explicitly.

I cannot simply use soft_assert as I need to wait for the message to show up.